### PR TITLE
netcdf: build both static and shared libraries

### PR DESCRIPTION
**Pull request by jeroen**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
-----

Adds an option to produce static library of netcdf instead of a dynamic library. We really need this to build our application. No side effects on standard installation. Thank you for considering.

### CODE REVIEW

File: Formula/netcdf.rb [hunk 1]
```diff
@@ -37,7 +39,11 @@ class Netcdf < Formula
   def install
     ENV.deparallelize
 
-    common_args = std_cmake_args << "-DBUILD_SHARED_LIBS=ON"
```
Comment (DomT4) [inline Formula/netcdf.rb]:
```ruby
common_args = std_cmake_args

if build.with? "static"
  common_args << "-DBUILD_SHARED_LIBS=OFF"
else
  common_args << "-DBUILD_SHARED_LIBS=ON"
end
```

File: Formula/netcdf.rb [hunk 2]
```diff
@@ -85,11 +91,13 @@ def install
     end
 
     # SIP causes system Python not to play nicely with @rpath
-    %w[libnetcdf-cxx4.dylib libnetcdf_c++.dylib].each do |f|
-      macho = MachO.open("#{lib}/#{f}")
-      macho.change_dylib("@rpath/libnetcdf.11.dylib",
-                         "#{lib}/libnetcdf.11.dylib")
-      macho.write!
+    if build.without? "static"
```
Comment (DomT4) [inline Formula/netcdf.rb]: Non-default options don't get tested, so this is a bit pointless IMO.

**Comment by jeroen** [comment]
@DomT4 Thank you. I have removed the redundant condition from the test now.

**Comment by ilovezfs** [comment]
Rather than an option. Let's just build both. See `snappy`.

**Review by ilovezfs — CHANGES_REQUESTED** [review]
both please

**Comment by jeroen** [comment]
OK it took some work I finally got this to work. 

For whatever reason we cannot just call `make install` twice because running `make install` on this lib will invoke `make uninstall` first. Therefore on the second run we manually install the static file.

**Review by ilovezfs — CHANGES_REQUESTED** [review]
>For some reason we cannot just call make install twice because running the second make install will uninstall the first one. Therefore we just manually lib.install the static libs.

This needs to be reported upstream

**Comment by jeroen** [comment]
ok

**Comment by jeroen** [comment]
This works fine as is. I don't think upstream intends on changing this behavior.

**Comment by ilovezfs** [comment]
Did you report it?

**Comment by jeroen** [comment]
https://github.com/Unidata/netcdf-c/issues/496

**Comment by ilovezfs** [comment]
@jeroen yeah it would be cool if they can support doing both at once in a single `make install`, but, at minimum, if they are going to require us to call `make install` twice, the second `make install` shouldn't wipe out the libraries from the first `make install`.

**Comment by jeroen** [comment]
Right, I will address this to upstream as well so that they can have a look at it the next time they are revising the cmake configuration. 

However it may be a while until they have a chance to look at this, and then more time until the next release. Is there any chance we can move forward with this PR in the mean time? The current code seems like the obvious solution to add static libs to the build?

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by jeroen** [comment]
thanks!

**Comment by jeroen** [comment]
Merge? 🚀

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
Thanks again @jeroen! :white_check_mark:

**Pull request closed without merging**